### PR TITLE
add: Dockerfile for installing sample-platform

### LINKED ISSUE #3: Add Docker container for python code
We currently have these docker containers:
- nginx proxy
- let's encrypt companion
- mysql database

We need to add a container to run the python code, and which can auto-install the dependencies if the container is installed.

### PR DESCRIPTION
**Pull request by thealphadollar**
The PR adds:

- Dockerfile for sample-platform

The sample platform needs to be configured to use the DB and Nginx containers, and accordingly, env variables need to be set (#5).

fixes #3

### CODE REVIEW

File: docker-orchestration/sample-platform-python/install.sh [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+echo "Welcome to the CCExtractor platform installer for docker!"
+echo ""
+echo "-------------------------------"
+echo "|   Installing dependencies   |"
+echo "-------------------------------"
+echo ""
+echo "* Updating package list"
+apt-get update
+echo "* Installing nginx, python, pip, kvm, libvirt and virt-manager"
+apt-get -q -y install nginx python python-dev python3-libvirt libxslt1-dev libxml2-dev python-pip qemu-kvm libvirt-dev virt-manager mediainfo
```
Comment (canihavesomecoffee) [inline docker-orchestration/sample-platform-python/install.sh]: It seems wiser to me to move this to the Dockerfile (I think most of this installer can actually be moved into a dockerfile when building...)

File: docker-orchestration/sample-platform-python/install.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+mkdir -p "${sample_repository}/QueuedFiles"
+
+config_db_uri="mysql+pymysql://${db_user}:${db_user_password}@localhost:3306/${db_name}"
+
+echo "-------------------------------"
+echo "|      Finalizing install     |"
+echo "-------------------------------"
+
+echo "* Generating config file"
+echo "# Auto-generated configuration by install.sh
```
Comment (canihavesomecoffee) [inline docker-orchestration/sample-platform-python/install.sh]: I'd move this to outside of Docker (the generation) and then pass it in through a volume. Ansible can create this for example.

**Review by canihavesomecoffee — CHANGES_REQUESTED** [review]

File: docker-orchestration/sample-platform-python/install.sh (hunk 2 above)
Comment (canihavesomecoffee) [reply docker-orchestration/sample-platform-python/install.sh]: Refer to https://github.com/canihavesomecoffee/sample-platform-ansible/blob/master/configure_platform.yaml, where this config file generation is already handled.

If we pass that generated file through a volume, we can drop a whole bunch of code here and in the docker file (the args IIRC)

Comment (thealphadollar) [reply docker-orchestration/sample-platform-python/install.sh]: fixed.

File: docker-orchestration/sample-platform-python/install.sh (hunk 1 above)
Comment (thealphadollar) [reply docker-orchestration/sample-platform-python/install.sh]: fixed.

File: docker-orchestration/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
+FROM python:3.9-slim-buster
+
+RUN apt-get update \
+    && apt-get -q -y install git gcc nginx python3-libvirt libxslt1-dev libxml2-dev qemu-kvm libvirt-dev virt-manager mediainfo
```
Comment (canihavesomecoffee) [inline docker-orchestration/Dockerfile]: qemu-kvm, libvirt, virt-manager, nginx won't be needed anymore in the new setup

File: docker-orchestration/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+WORKDIR /var/www
+RUN git clone https://github.com/CCExtractor/sample-platform.git
```
Comment (canihavesomecoffee) [inline docker-orchestration/Dockerfile]: I wouldn't clone the SP in the docker build, but have it passed to the docker instance through a volume. That'll make it easier for us to update (I think)

File: docker-orchestration/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+WORKDIR /var/www/sample-platform/
+RUN python -m pip install --upgrade pip setuptools wheel \
+    && pip install -r requirements.txt
+
+RUN cp "install/platform" /etc/init.d/platform \
```
Comment (canihavesomecoffee) [inline docker-orchestration/Dockerfile]: It seems to me that we could have the command that's in the platform startup script as the container run script, and configure the container to restart as soon as it dies. That's a more "docker"-like way than using an initscript that I'm not sure will be running by itself if you restart the container.

**Review by canihavesomecoffee — CHANGES_REQUESTED** [review]

File: docker-orchestration/Dockerfile (hunk 5 above)
Comment (thealphadollar) [reply docker-orchestration/Dockerfile]: fixed via 32fb56d

File: docker-orchestration/Dockerfile (hunk 4 above)
Comment (thealphadollar) [reply docker-orchestration/Dockerfile]: fixed via 32fb56d

File: docker-orchestration/Dockerfile (hunk 3 above)
Comment (thealphadollar) [reply docker-orchestration/Dockerfile]: fixed via 32fb56d

**Review by canihavesomecoffee — APPROVED** [review]

**Pull request merged**